test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -61,38 +61,48 @@ type Grabber struct {
 	grabFromKubelets                  bool
 	grabFromScheduler                 bool
 	grabFromClusterAutoscaler         bool
-	masterName                        string
-	registeredMaster                  bool
+	kubeScheduler                     string
+	kubeControllerManager             string
 	waitForControllerManagerReadyOnce sync.Once
 }
 
 // NewMetricsGrabber returns new metrics which are initialized.
 func NewMetricsGrabber(c clientset.Interface, ec clientset.Interface, kubelets bool, scheduler bool, controllers bool, apiServer bool, clusterAutoscaler bool) (*Grabber, error) {
-	registeredMaster := false
-	masterName := ""
-	nodeList, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+
+	kubeScheduler := ""
+	kubeControllerManager := ""
+
+	regKubeScheduler := regexp.MustCompile("kube-scheduler-.*")
+	regKubeControllerManager := regexp.MustCompile("kube-controller-manager-.*")
```

Comment:
i remembered a discussion around the fact that this is not a reliable way to determine if the namespace has the components in question, because someone might have decided to name them differently in their custom deployment.

but i'd argue that it's much better compared to using the deprecated "master" function and we are not going to return errors for them, but just skip the metrics gathering.